### PR TITLE
Modernize wpcom-proxy-request

### PR DESCRIPTION
**Pull request by sgomes**
This PR modernizes `wpcom-proxy-request` in several ways:
- Switches to building through `calypso-build`
- Updates `debug` dependency
- Reorganizes both source and package directories to match `calypso-build` expectations

It also switches over to using this package in Calypso and `package/data-stores`. It keeps the package private in the meantime, though, in case we want to make further changes before the next major version release as an independent npm package.

This PR should remove the last dependency on `babel-runtime` (as opposed to the newer `@babel/runtime`, which we already use), and it should relegate usage of `core-js@2` to devdocs, leaving the main entry points with just `core-js@3`.

The size savings are smaller than I hoped for (~7KB compressed), but still very desirable.

#### Changes proposed in this Pull Request

* Modernize `wpcom-proxy-request`
* Switch to the in-repo package in Calypso and `package/data-stores`
* Fix some `wpcom-proxy-request` lint errors

#### Testing instructions

Pretty much all API requests go through `wpcom-proxy-request`, so some casual testing of the live branch should be enough to ensure that nothing went horribly wrong with this change.

### CODE REVIEW

**Comment by sgomes** [comment]
The matticbot numbers are incorrect. This PR does not add 67KB per entry point.

**Comment by sgomes** [comment]
Rebased on master and manually fixed `package-lock.json` after several unsuccessful attempts. Rebasing when changing dependencies is a real pain 😞

**Comment by sgomes** [comment]
Looks like #39410 changed everything 😞 Redoing all my changes and trying not to miss anything...

**Comment by sirreal** [comment]
The package-lock indicates this is pulling in a unique `debug` version. Any idea where that's coming from?

```
		"wpcom-proxy-request": {
			"version": "file:packages/wpcom-proxy-request",
			"requires": {
				"component-event": "^0.1.4",
				"debug": "^4.1.1",
				"progress-event": "^1.0.0",
				"uid": "^0.0.2",
				"wp-error": "^1.3.0"
			},
			"dependencies": {
				"debug": {
					"version": "3.2.6",
					"resolved": "https://registry.npmjs.org/debug/-/debug-3.2.6.tgz",
					"integrity": "sha512-mel+jf7nrtEl5Pn1Qx46zARXKDpBbvzezse7p7LqINmdoIk8PYP5SySaxEmYv6TZ0JyEKA1hsCId6DIhgITtWQ==",
					"requires": {
						"ms": "^2.1.1"
					}
				}
			}
		},
```

File: packages/wpcom-proxy-request/src/index.js [hunk 1]
```diff
@@ -98,13 +98,13 @@ debug( 'using "origin": %o', origin );
  * takes care of WordPress.com user authentication (via the currently
  * logged-in user's cookies).
  *
- * @param {Object} originalParams - request parameters
+ * @param {object} originalParams - request parameters
  * @param {Function} [fn] - callback response
- * @return {XMLHttpRequest} XMLHttpRequest instance
- * @api public
+ * @returns {window.XMLHttpRequest} XMLHttpRequest instance
+ * @public
```
Comment (sirreal) [inline packages/wpcom-proxy-request/src/index.js]: In an ES module that exports the public functions I'd vote to remove all these `@public`/`@private` tags.

```suggestion
```

**Comment by sgomes** [comment]
> The package-lock indicates this is pulling in a unique `debug` version. Any idea where that's coming from?

Looks like an `npm dedupe` was needed here? Or maybe it's an `npm` bug; the dependency shouldn't have persisted if I changed the versions. So buggy...

**Review by sirreal — APPROVED** [review]
Changes look good and smoke testing doesn't indicate any problems. I left a minor suggestion to remove `@public`/`@private` JSDoc tags, and I wonder if we can fix what looks like a stale `debug@3` dependency.

**Comment by sgomes** [comment]
Manually removed.

File: packages/wpcom-proxy-request/src/index.js (hunk 1 above)
Comment (sgomes) [reply packages/wpcom-proxy-request/src/index.js]: Done!

**Comment by sgomes** [comment]
Thank you for the review, @sirreal! 👍

**Pull request merged**